### ISSUE TITLE
gnome3 enables networkmanager, which disables useDHCP, which breaks initrd network

### ISSUE THREAD

**Issue by andersk** [NixOS/nixpkgs#50832]
## Issue description

On a system with initrd networking configured with DHCP, if you change `services.xserver.desktopManager.gnome3.enable` to `true`, suddenly

* nixos/modules/services/x11/desktop-managers/gnome3.nix defaults `networking.networkmanager.enable` to `true`,
* so nixos/modules/services/networking/networkmanager.nix forces `networking.useDHCP` to `false`,
* so nixos/modules/system/boot/initrd-network.nix refuses to run `udhcpc`,
* so the initrd environment has no networking.

You might agree that this is a surprising consequence of enabling a desktop environment.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.81, NixOS, 18.09.1373.50fb6820759 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(root): `"nixos-18.09.1373.50fb6820759"`
 - channels(anders): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by volth** [comment]
```.useDHCP``` can be set per-interface: ```networking.interface.$name.useDHCP = true```

and ```initrd-network``` understands it since https://github.com/NixOS/nixpkgs/pull/46459 (not sure if it was backported to ```18.09```)

```networking.useDHCP = true``` means that interface names are unknown at the configuration time and DHCP is to be tried on all the interfaces.

**Comment by andersk** [comment]
@volth #46459 was not backported to 18.09. (The scenario I’m describing really happened to my 18.09 system, it’s not hypothetical.)

**Comment by jahkeup** [comment]
I agree, NetworkManager enabled as a side effect of enabling a desktop manager is surprising to say the least. To me, this was especially confusing as I was enabling and configuring dhcpcd and configuring for appropriate interfaces - which was working up until I enabled gnome3 for another user.

I'm not positive how exactly, but it would seem as though the config's intent to configure another network "utility/daemon/manager" could result in NetworkManager /not/ being enabled.

**Comment by jaen** [comment]
Just came across that when trying to add SSH LUKS unlocking to a machine that uses NetworkManager.
